crates/teloxide/examples/deep_linking.rs

```diff
@@ -0,0 +1,133 @@
+use dptree::{case, deps};
+use teloxide::{
+    dispatching::dialogue::{self, InMemStorage},
+    macros::BotCommands,
+    prelude::*,
+    types::Me,
+};
+
+pub type MyDialogue = Dialogue<State, InMemStorage<State>>;
+pub type HandlerResult = Result<(), Box<dyn std::error::Error + Send + Sync>>;
+
+#[derive(Clone, PartialEq, Debug, Default)]
+pub enum State {
+    #[default]
+    Start,
+    WriteToSomeone {
+        id: i64,
+    },
+}
+
+#[derive(BotCommands, Clone, Debug)]
+#[command(rename_rule = "lowercase")]
+pub enum StartCommand {
+    #[command()]
+    Start(String), /* Because deep linking (links like https://t.me/some_bot?start=123456789)
+                    * is the same as sending "/start 123456789",
+                    * we can treat it as just an argument to a command
+                    *
+                    * https://core.telegram.org/bots/features#deep-linking */
+}
+
+#[tokio::main]
+async fn main() {
+    pretty_env_logger::init();
+    log::info!("Starting dialogue bot...");
```

Comment:
Maybe, "starting \"deep linking\" example.."?

Reply:
Oh, sorry, yes, i just copied it in